velox/substrait/VeloxToSubstraitExpr.cpp

```diff
@@ -0,0 +1,383 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/substrait/VeloxToSubstraitExpr.h"
+
+#include "velox/expression/ControlExpr.h"
+
+namespace facebook::velox::substrait {
+
+void VeloxToSubstraitExprConvertor::toSubstraitExpr(
+    const std::shared_ptr<const ITypedExpr>& vExpr,
+    RowTypePtr vPreNodeOutPut,
+    ::substrait::Expression* sExpr) {
+  if (std::shared_ptr<const ConstantTypedExpr> vConstantExpr =
+          std::dynamic_pointer_cast<const ConstantTypedExpr>(vExpr)) {
+    toSubstraitLiteral(vConstantExpr->value(), sExpr->mutable_literal());
+    return;
+  }
+  if (auto vCallTypeExpr =
+          std::dynamic_pointer_cast<const CallTypedExpr>(vExpr)) {
+    std::shared_ptr<const Type> vExprType = vCallTypeExpr->type();
+    std::vector<std::shared_ptr<const ITypedExpr>> vCallTypeInputs =
+        vCallTypeExpr->inputs();
+    std::string vCallTypeExprFunName = vCallTypeExpr->name();
+    // different by function names.
+    if (vCallTypeExprFunName == exec::kIf) {
```

Comment:
could we move this entire part of the code to another `toSubstraitExpr()` helper function to make it more modular?